dxr/plugins/xpidl/indexers.py

```diff
@@ -0,0 +1,48 @@
+from os.path import join
+from xpidl.xpidl import IDLParser, IDLError
+
+import dxr.indexers
+from dxr.indexers import iterable_per_line, with_start_and_end, split_into_lines
+from dxr.plugins.xpidl.filters import PLUGIN_NAME
+from dxr.plugins.xpidl.visitor import IdlVisitor
+
+
+class FileToIndex(dxr.indexers.FileToIndex):
+    def __init__(self, path, contents, plugin_name, tree):
+        super(FileToIndex, self).__init__(path, contents, plugin_name, tree)
+        self.temp_folder = join(self.tree.temp_folder, 'plugins', PLUGIN_NAME)
+        self.parser = IDLParser(self.temp_folder)
+        self._idl = None
+        self._had_idl_exception = False
+
+    @property
+    def idl(self):
+        """Parse the IDL file and resolve dependencies. If successful, return an IdlVisitor
+        object which has visited the AST and has refs and needles ready for ES. Otherwise,
+        on exception, return None."""
+
+        # Don't try again if we already excepted.
+        if not self._idl and not self._had_idl_exception:
+            try:
+                self._idl = IdlVisitor(self.parser, self.contents, self.contents.splitlines(True),
+                                       self.path, self.absolute_path(),
+                                       self.plugin_config.include_folders,
+                                       self.plugin_config.header_path, self.tree)
+            except IDLError:
+                self._had_idl_exception = True
```

Comment:
Meh, if we can't get our extensions right, let's leave it alone. Later, maybe we can come back and have it actually look at the contents of the file to determine what kind of IDL file it is.